snap/gadget.go

```diff
@@ -249,3 +490,101 @@ func (e *editionNumber) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	*e = editionNumber(u)
 	return nil
 }
+
+// GadgetSize describes the size of a structure item or an offset within the
+// structure.
+type GadgetSize uint64
```

Comment:
It's cluttering the `snap` namespace a bit now, but once I introduce the `gadget` package, it would become `gadget.Size`, where it could probably stay exported. What do you think?

Reply:
it's fine (in that world)